pkg/imageallowrules/imageallowrules.go

```diff
@@ -0,0 +1,118 @@
+package imageallowrules
+
+import (
+	"context"
+	"fmt"
+
+	v1 "github.com/acorn-io/acorn/pkg/apis/internal.acorn.io/v1"
+	"github.com/acorn-io/acorn/pkg/cosign"
+	"github.com/acorn-io/acorn/pkg/images"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+	ociremote "github.com/sigstore/cosign/pkg/oci/remote"
+	"github.com/sirupsen/logrus"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type ErrImageNotAllowed struct {
+	Rule  string
+	Image string
+}
+
+func (e *ErrImageNotAllowed) Error() string {
+	return fmt.Sprintf("image %s is not allowed by rule %s", e.Image, e.Rule)
+}
+
+func (e *ErrImageNotAllowed) Is(target error) bool {
+	_, ok := target.(*ErrImageNotAllowed)
+	return ok
+}
+
+// CheckImageAllowed checks if the image is allowed by the ImageAllowRules in the same namespace as the AppInstance
+// @return string denoting the reason for *not* allowing the image (empty string if image is allowed)
```

Comment:
Nit: this function doesn't return a string anymore

Reply:
Yup, removed that :)